src/dom-to-react/index.js

```diff
@@ -82,7 +108,7 @@ class Dom2React {
           case 'tr':
             return null;
           default:
-            return node.nodeValue.toString();
+            return node.nodeValue.trim().toString();
```

Comment:
I was thinking about this in the beginning but in the end decided not to trim the text for cases like this:
``` jsx
<div class="some-message">
  Some message before a <span>span</span>
</div>
```
when trimming the textnode "Some message before a " it would result in "aspan" in the readable/rendered text

Reply:
Good point. I'll update to only trim on nodes that should not have spaces like colspan rather than all nodes.

Reply:
Do you have an example where a node contains text but needs to be trimmed?
I currently can't think of a case where this might be required.

Reply:
i have a table here which will trigger an error in react unless i trim().

```
<table style="width: 99.9361%;">
    <colgroup>
        <col style="width: 7.8055%;">
        <col style="width: 15.4191%;">
        <col style="width: 8.50928%;">
        <col style="width: 11.3884%;">
        <col style="width: 27.7031%;">
        <col style="width: 8.57326%;">
        <col style="width: 20.4095%;">
        <col style="width: 0.191939%;">
    </colgroup>
    <tbody>
    <tr>
        <th><h2>heading 2</h2></th>
        <th><h2>heading 2</h2></th>
        <th colspan="1"><h2>heading 2</h2></th>
        <th><h2>heading 2</h2></th>
        <th colspan="1"><h2>heading 2</h2></th>
        <th colspan="1"><h2>heading 2</h2>
        </th>
        <th><h2>heading 2</h2>
        </th>
    </tr>
    <tr>
        <td colspan="1"><br></td>
        <td colspan="1"><br></td>
        <td colspan="1"><br></td>
        <td colspan="1"><br></td>
        <td colspan="1"><br></td>
        <td colspan="1"><br></td>
        <td colspan="1"><br></td>
    </tr>
    <tr>
        <td data-highlight-colour="blue"><h3>heading 3</h3></td>
        <td colspan="1">text</td>
        <td colspan="6"><br><br><br></td>
    </tr>
    </tbody>
</table>
```

Reply:
Ah, I got it.

this is more like with the <tr> which may not contain a textnode at all.

I think it would be enough to add those tags together with
``` javascript
case 'tr':
case 'colgroup':
case 'tbody':
...
  return null;
```
as all these nodes are not allowed to contain textnodes at all.

The best would be to find a list of all html-nodes which are not allowed to contain textnodes

I can currently think of  `iframe, thead, tbody, tfoot, ul, ol, dl, br, colgroup, col, table, select, input`

I think for now I'd be fine with these.

If this would solve your problem, I'd add these to the list for now.

Reply:
I implemented the list of nodes which should not contain non-whitespace textnodes.
[Textnodes with just whitespace inside now get ignored/removed](https://github.com/diva-e/dom-to-react/blob/3a4aa502ed4a2dc2e00b87f8a955246894133b5a/src/dom-to-react/index.js#L86).

I'll publish another release once the attribute-names (other discussion) have been resolved.